nca/NetworkConfig/NetworkConfig.py

```diff
@@ -172,17 +172,33 @@ def get_affected_pods(self, is_ingress, layer_name):
 
         return affected_pods
 
-    def get_referenced_ip_blocks(self):
+    def _check_for_excluding_ipv6_addresses(self, exclude_ipv6):
         """
+        checks and returns if to exclude non-referenced IPv6 addresses from the config
+        Excluding the IPv6 addresses will be enabled if the exclude_ipv6 param is True and
+        IPv6 addresses in all the policies of the config (if existed) were added automatically by the parser
+        and not referenced by user
+        :param bool exclude_ipv6: indicates if to exclude ipv_6 non-referenced addresses
+        :rtype bool
+        """
+        for policy in self.policies_container.policies.values():
+            if policy.has_ipv6_addresses:  # if at least one policy has referenced ipv6 addresses, ipv6 will be included
+                return False
+        return exclude_ipv6  # getting here means all policies didn't reference ipv6, excluding depends on the given flag
```

Comment:
if the given flag `exclude_ipv6` is `false`, then no need to iterate the policies